File: src/main/java/com/dude/dms/backend/repositories/PlainTextRuleRepository.java
package com.dude.dms.backend.repositories;

import com.dude.dms.backend.data.rules.PlainTextRule;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PlainTextRuleRepository extends JpaRepository<PlainTextRule, Long> {

    List<PlainTextRule> findByActive(boolean active);
}

File: src/main/kotlin/com/dude/dms/backend/repositories/PlainTextRuleRepository.kt
package com.dude.dms.backend.repositories

import com.dude.dms.backend.data.Tag
import com.dude.dms.backend.data.filter.PlainTextRule
import org.springframework.data.jpa.repository.JpaRepository

interface PlainTextRuleRepository : JpaRepository<PlainTextRule, Long> {

    fun findByTags(tag: Tag): Set<PlainTextRule>

    fun countByTags(tag: Tag): Long
}